### PR TITLE
ref(ci): Adding a step to collect test data for backend tests

### PR DESCRIPTION
**Pull request by IanWoodard**
Building off of #68263. Adding usages of [action-collect-test-data](https://github.com/getsentry/action-collect-test-data) to start collecting data about our backend tests.

### CODE REVIEW

File: .github/workflows/acceptance.yml [hunk 1]
```diff
@@ -45,6 +45,9 @@ jobs:
     name: acceptance
     runs-on: ubuntu-22.04
     timeout-minutes: 30
+    permissions:
+      contents: read
```
Comment (Jeffreyhung) [inline .github/workflows/acceptance.yml:49]:
```suggestion
```
By default, permission to read contents is granted to the token, so no need to specify them.
https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

File: .github/workflows/backend.yml [hunk 2]
```diff
@@ -261,6 +277,16 @@ jobs:
...
 
+      - name: Collect test data
+        if: ${{ !cancelled() }}
+        uses: getsentry/action-collect-test-data@c0a2fd3f79d3867e2c8b486ccb1c4bf925477742 # v0.3.0
+        with:
+          path: .artifacts/pytest.monolith-dbs.json
+          gcs_path: ${{ secrets.COLLECT_TEST_DATA_GCS_BUCKET }}/pytest/${{ github.repository }}/${{ github.workflow }}/${{ github.job }}/${{ github.repository_id }}/${{ github.run_id }}/${{ github.run_attempt }}
+          gcp_project_id: ${{ secrets.COLLECT_TEST_DATA_GCP_PROJECT_ID }}
+          workload_identity_provider: ${{ secrets.SENTRY_GCP_DEV_WORKLOAD_IDENTITY_POOL }}
+          service_account_email: ${{ secrets.COLLECT_TEST_DATA_SERVICE_ACCOUNT_EMAIL }}
```
Comment (asottile-sentry) [inline .github/workflows/backend.yml:288]: this should probably just be part of our collect-artifacts step

**Review by asottile-sentry — APPROVED** [review]

File: .github/workflows/acceptance.yml (hunk 1 above)
Comment (IanWoodard) [reply .github/workflows/acceptance.yml:49]: It doesn't hurt to be explicit

**Pull request merged**